examples/newsreaderExample/src/main/java/io/realm/examples/newsreader/model/network/NYTimesDataLoader.java

```diff
@@ -0,0 +1,142 @@
+            "health",
+            "sports",
+            "arts",
+            "fashion",
+            "dining",
+            "travel",
+            "magazine",
+            "realestate"
+    };
+
+    private NYTimesService nyTimesService;
+    private SimpleDateFormat inputDateFormat = new SimpleDateFormat("yyyy-MM-d'T'HH:mm:ssZZZZZ");
+    private SimpleDateFormat outputDateFormat = new SimpleDateFormat("MM-dd-yyyy");
+    private String apiKey;
+    private Realm realm;
+    private BehaviorSubject<Boolean> networkInUse;
+
+    public NYTimesDataLoader() {
+        Retrofit retrofit = new Retrofit.Builder()
+                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
+                .addConverterFactory(JacksonConverterFactory.create())
+                .baseUrl("http://api.nytimes.com/")
+                .build();
+        nyTimesService = retrofit.create(NYTimesService.class);
+    }
+
+    public void loadAllData(final Realm realm, String apiKey, BehaviorSubject<Boolean> networkUsage) {
+        this.apiKey = apiKey;
+        this.realm = realm;
+        this.networkInUse = networkUsage;
+        loadNextSection(0);
+    }
+
+    // Load all sections one by one.
+    private void loadNextSection(final int sectionIndex) {
+        networkInUse.onNext(true);
+        nyTimesService.topStories(nytSections[sectionIndex], apiKey)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new Action1<NYTimesResponse<List<NYTimesStory>>>() {
+                    @Override
+                    public void call(NYTimesResponse<List<NYTimesStory>> response) {
+                        Timber.d("Success - Data received: %s", nytSections[sectionIndex]);
+                        processAndAddData(realm, response.results);
+
+                        if (sectionIndex < nytSections.length - 1) {
+                            loadNextSection(sectionIndex + 1);
+                        } else {
+                            networkInUse.onNext(false);
+
+                        }
+                    }
+                }, new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        networkInUse.onNext(false);
+                        Timber.d("Failure: Data not loaded: %s - %s", nytSections[sectionIndex], throwable.toString());
+                    }
+                });
+    }
+
+    // Converts data into a usable format and save it to Realm
+    private void processAndAddData(final Realm realm, final List<NYTimesStory> stories) {
+        if (stories.isEmpty()) return;
+
+        realm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                for (NYTimesStory story : stories) {
+                    Date parsedPublishedDate = inputDateFormat.parse(story.getPublishedDate(), new ParsePosition(0));
```

Comment:
why is this transformation not in the `Model` or `Repository`?, why not using a `JsonDeserializer` that deserialise Dates etc. 
also you're not using the index of `ParsePosition`